bijection-util/src/main/scala/com/twitter/bijection/twitter_util/JavaFutureConverter.scala

```diff
@@ -0,0 +1,81 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package com.twitter.bijection.twitter_util
+
+import java.util.concurrent.{ Future => JFuture }
+
+import com.twitter.util._
+
+/**
+ * Base class for converting java futures to twitter futures
+ */
+abstract class JavaFutureConverter {
+  def apply[T](javaFuture: JFuture[T]): Future[T]
+}
+
+/**
+ * Converter based on a [[FuturePool]] which will create one thread per future.
+ * To favor if there aren't too many futures to convert and one cares about latency.
+ * @param mayInterruptIfRunning whether or not the initial java future can be interrupted if it's
+ *                              running
+ */
+class FuturePoolJavaFutureConverter(mayInterruptIfRunning: Boolean) extends JavaFutureConverter {
+  override def apply[T](javaFuture: JFuture[T]): Future[T] = {
+    val f = FuturePool.unboundedPool { javaFuture.get() }
+    val p = Promise.attached(f)
+    p.setInterruptHandler { case NonFatal(e) =>
+      if (p.detach()) {
+        f.raise(e)
+        javaFuture.cancel(mayInterruptIfRunning)
+      }
+    }
+    p
+  }
+}
+
+/**
+ * Converter based on a [[Timer]] which will create a task which will check every
+ * <code>checkFrequency</code> if the java future is completed, one thread will be used for every
```

Comment:
the threading model is determined by the timer, not be the implementation.  most timers have a small, bounded thread pool, and they typically default to a size 1 thread pool.

it might also be worth noting that the minimum `checkFrequency` will be affected by the granularity of the timer.  If you're using a timer with tick granularity of 10ms, setting a `checkFrequency` of 0.5ms will not do you a ton of good.